ibis/expr/api.py

```diff
@@ -4193,22 +4193,57 @@ def mutate(table, exprs=None, **mutations):
         for name, expr in sorted(mutations.items(), key=operator.itemgetter(0))
     )
 
+    # The below logic computes the mutation node exprs by splitting the
+    # assignment exprs into two disjoint sets:
+    # 1) overwriting_cols_to_expr, which maps a column name to its expr
```

Comment:
You probably want to specify (1) when does a column name maps to a expr (the first column that was overwritten) and (2) when does a column name maps to None (the second+ column that was overwritten, destructure only) (3) when does a column name doesn't exist (column was not overwritten)

Reply:
I would also make clear that this maps a column name in the *original table* to ...

Reply:
Added a more descriptive comment that captures these points explicitly